webapp/src/plugin.jsx

```diff
@@ -37,6 +41,7 @@ const setupUILater = (registry, store) => async () => {
         registry.registerWebSocketEventHandler(`custom_${PluginId}_connect`, handleConnectChange(store));
         registry.registerWebSocketEventHandler(`custom_${PluginId}_disconnect`, handleConnectChange(store));
         registry.registerWebSocketEventHandler(`custom_${PluginId}_instance_status`, handleInstanceStatusChange(store));
+        haveSetUpUI = true;
```

Comment:
This seems to still be theoretically race-prone. Should the flag semantics be `setupUIStarted`, and set immediately after checking it?

Reply:
@levb Yah, that makes sense. I just didn't want it to fail and then have nothing. But your worry is more important.